openprescribing/matrixstore/sql_functions.py

```diff
@@ -5,13 +5,20 @@
 
 
 class MatrixSum(object):
+    """
+    Provides an optimised (for our use case) routine for summing matrices
+
+    Note the `step` and `finalize` methods are what allow this class to be used
+    as an SQLite custom aggregation function.
+    """
 
     accumulator = None
 
     def step(self, value):
-        if value is None:
-            return
-        matrix = deserialize(value)
+        if value is not None:
+            self.add(deserialize(value))
+
+    def add(self, matrix):
```

Comment:
This did cross my mind. But I feel like it might end up as a case of syntactic sugar rotting semantic teeth, as the `MatrixSum` object doesn't otherwise behave like a matrix. In particular, you have to call `value()` on it to get a result back.